internal/provider/helpers/marshal.go

```diff
@@ -5,9 +5,10 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/diag"
 )
 
-// SafeUnmarshal is a helper function for safely unmarshalling a JSON object by checking if
-// it is null before attempting to unmarshal it. This should always be used for optional attributes.
-func SafeUnmarshal(attribute jsontypes.Normalized) (map[string]interface{}, diag.Diagnostics) {
+// UnmarshalOptional attempts to unmarshal an optional attribute. If it is null, it returns an empty
+// map[string]interface{} and no diagnostics. If it is not null, it attempts to unmarshal it and returns
+// any diagnostics.
+func UnmarshalOptional(attribute jsontypes.Normalized) (map[string]interface{}, diag.Diagnostics) {
```

Comment:
This is a cosmetic change, but hopefully better communicates the purpose of this function (528475b).